Marlin/ultralcd.cpp

```diff
@@ -1589,13 +1589,14 @@ void kill_screen(const char* lcd_msg) {
       // Utility to go to the next mesh point
       inline void _manual_probe_goto_xy(float x, float y) {
         #if MANUAL_PROBE_HEIGHT > 0
-          line_to_z(LOGICAL_Z_POSITION(Z_MIN_POS) + MANUAL_PROBE_HEIGHT);
+          const float prev_z = current_position[Z_AXIS];
+          line_to_z(current_position[Z_AXIS] + MANUAL_PROBE_HEIGHT);
```

Comment:
Assuming the bed isn't tilted at an exaggerated angle, this is more appropriate:
```cpp
const float prev_z = current_position[Z_AXIS];
line_to_z(LOGICAL_Z_POSITION(Z_MIN_POS) + MANUAL_PROBE_HEIGHT);
```